internal/test/framework/framework.go

```diff
@@ -0,0 +1,198 @@
+package framework
+
+import (
+	"fmt"
+	"k8s.io/client-go/tools/clientcmd"
+	"log"
+	"os"
+	"time"
+
+	"github.com/masterzen/winrm"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/cloudprovider"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/types"
+	"golang.org/x/crypto/ssh"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// user for the Windows node created.
+	// TODO: remove this hardcoding to any user.
+	user = "Administrator"
+	// winrm port to be used
+	winRMPort = 5986
+	// remotePowerShellCmdPrefix holds the powershell prefix that needs to be prefixed to every command run on the
+	// remote powershell session opened
+	remotePowerShellCmdPrefix = "powershell.exe -NonInteractive -ExecutionPolicy Bypass "
+)
+
+// TestFramework holds the info to run the test suite.
+// This is not clean
+type TestFramework struct {
+	// Credentials to access the Windows VM created
+	Credentials *types.Credentials
+	// WinrmClient to access the Windows VM created
+	WinrmClient *winrm.Client
+	// remoteDir is the directory to which files will be transferred to, on the Windows VM
+	RemoteDir string
+	// SSHClient contains the ssh client information to access the Windows VM via ssh
+	SSHClient *ssh.Client
+	// CloudProvider holds the information related to cloud provider
+	cloudProvider cloudprovider.Cloud
+	// k8sclientset is the kubernetes clientset we will use to query the cluster's status
+	K8sclientset *kubernetes.Clientset
+}
```

Comment:
Do all the fields in the struct need to exported i.e. capitalized?

Reply:
Not all of them, for example `cloudProvider` doesn't need to be exported. But as I told at https://github.com/openshift/windows-machine-config-operator/pull/103/files#diff-5465ccc52b3a24461a98f029910cf60aR29, this is not clear, ideally we should have getters and setters around this